### PR TITLE
Add add_flex action and error logging for possible environment variable typos 

### LINKED ISSUE #161: feature request: support openpathlist for option type
`openpathlist` option type as an alternative to the existing `pathlist` option type, which allows prepending/appending rather than always overwriting

export EASYBUILD_ROBOT_PATHS=:/custom  # append to existing value
export EASYBUILD_ROBOT_PATHS=/custom:  # prepend to existing value
export EASYBUILD_ROBOT_PATHS=/custom  # overwrite existing value (like for 'pathlist'-typed option)

Would be useful in the context of EasyBuild to avoid overwriting `--robot-paths` default value.

cc @dominikborkowski

### PR DESCRIPTION
**Pull request by stdweird**
fixes #161

### CODE REVIEW

**Comment by stdweird** [comment]
@boegel this should be sufficient for the EB issues you wanted

File: lib/vsc/utils/generaloption.py [hunk 1]
```diff
@@ -106,6 +106,22 @@ def check_str_list_tuple(option, opt, value):
         return klass(split)
 
 
+def get_empty_add_flex(allvalues):
+    """Return the empty element for add_flex action for allvalues"""
+    empty = None
+    emptyisNone = False # in case None is the empty value
```
Comment (boegel) [inline lib/vsc/utils/generaloption.py]: drop silly-named and not even used `emptisNone`

Comment (stdweird) [inline lib/vsc/utils/generaloption.py]: it's there for future extensions

Comment (boegel) [inline lib/vsc/utils/generaloption.py]: extensions of what?

don't include code you're not actually using, it looks silly

**Comment by boegel** [comment]
still fixes and moar tests via https://github.com/stdweird/vsc-base/pull/4 (also fixing the merge conflict on the vsc-base version)

**Comment by boegel** [comment]
@stdweird: I ran into a potential issue when writing up the unit tests for EB's `--robot-paths` that relies on this, see https://github.com/hpcugent/easybuild-framework/pull/1252/files#r27655849

**Comment by stdweird** [comment]
@boegel rereview or merge

**Comment by boegel** [comment]
ready to go in imho, but I'd like to see #165 and #166 merged first, to avoid more version bumps

**Comment by stdweird** [comment]
@boegel bug is pretty stupid, not sure about the fix yet

**Comment by boegel** [comment]
"Would you like to know more?" *clicks yes*

File: setup.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def remove_bdist_rpm_source_file():
 
 PACKAGE = {
     'name': 'vsc-base',
-    'version': '2.1.2',
+    'version': '2.2.0',
```
Comment (boegel) [inline setup.py]: this wil need to be bumped again, cfr #168

**Comment by boegel** [comment]
fix for the bug that is causing this in https://github.com/stdweird/vsc-base/pull/6/files

**Comment by boegel** [comment]
Going in (finally), thanks @stdweird for looking into this!

**Pull request merged**